### PR TITLE
Support visualization & analysis for shorting

### PR DESCRIPTION
**Pull request by hieuh25**

### CODE REVIEW

File: tradeexecutor/visual/single_pair.py [hunk 1]
```diff
@@ -70,7 +70,8 @@ def get_position_hover_text(p: TradingPosition) -> str:
         ]
 
     text += [
-        f"Entry price: {first_trade.planned_mid_price:.2f} USD (mid price)",
+        # TODO: not sure why this failed for short position
+        # f"Entry price: {first_trade.planned_mid_price:.2f} USD (mid price)",
```
Comment (hieuh25) [inline tradeexecutor/visual/single_pair.py:74]: @miohtama I can look closer at this tomorrow, but do you have any idea why `planned_mid_price` is None here?

File: tradeexecutor/analysis/trade_analyser.py [hunk 2]
```diff
@@ -846,7 +846,10 @@ def expand_timeline(
     def expander(row):
         position: TradingPosition = row["position"]
         # timestamp = row.name  # ???
-        pair_id = position.pair.internal_id
+        if position.is_short():
+            pair_id = position.pair.underlying_spot_pair.internal_id
```
Comment (miohtama) [inline tradeexecutor/analysis/trade_analyser.py:850]: You can call `position.pair.get_pricing_pair()` here so you do not need `if`

File: tradeexecutor/visual/utils.py [hunk 3]
```diff
@@ -44,7 +44,12 @@ def export_trade_for_dataframe(p: Portfolio, t: TradeExecution) -> dict:
 
     position = p.get_position_by_id(t.position_id)
 
-    price_prefix = f"{t.pair.base.token_symbol} / USD"
+    if t.is_short():
+        base_token_symbol = t.pair.base.underlying.token_symbol
```
Comment (miohtama) [inline tradeexecutor/visual/utils.py:48]: Same as above

File: tradeexecutor/visual/utils.py [hunk 4]
```diff
@@ -146,8 +151,12 @@ def export_trades_as_dataframe(
     data = []
 
     for t in portfolio.get_all_trades():
-        if pair_id is not None and t.pair.internal_id != pair_id:
-            continue
+        if pair_id is not None:
+            if t.is_short():
+                if t.pair.underlying_spot_pair.internal_id != pair_id:
```
Comment (miohtama) [inline tradeexecutor/visual/utils.py:156]: Same as above

File: tradeexecutor/visual/utils.py [hunk 5]
```diff
@@ -279,9 +288,15 @@ def visualise_trades(
 def get_all_positions(state: State, pair_id):
     """Get all positions for a given pair"""
     assert type(pair_id) == int
-    positions = [
-        p for p in state.portfolio.get_all_positions() if p.pair.internal_id == pair_id
-    ]
+
+    positions = []
+    for p in state.portfolio.get_all_positions():
+        if p.is_long():
```
Comment (miohtama) [inline tradeexecutor/visual/utils.py:294]: Should be if `p.is_spot()`

**Review by miohtama — CHANGES_REQUESTED** [review]

**Pull request merged**